frontend/src/pages/AllListings.tsx

```diff
@@ -1,31 +1,71 @@
+import RecommendedCarousel from 'src/components/Recommended/RecommendedCarousel'
 
 export interface IAllListingsProps { }
 
 export default function AllListings(props: IAllListingsProps) {
-  const [listings, setListings] = React.useState([]);
-  const { sendRequest } = useHttpClient();
-
-  // Retrieving listings
-  React.useEffect(() => {
-    const getSellerItems = async () => {
-      let items = await sendRequest(LISTING, "GET", {});
-      items = buildListings(items);
-      setListings(items);
-    };
-    getSellerItems();
-  }, [sendRequest]);
-
-  return (
-    <React.Fragment>
-      <PageHeader headerTitle="Listing">
-        <ListingGrid listings={listings} />
-      </PageHeader>
-    </React.Fragment>
-  );
+    const [listings, setListings] = React.useState([]);
+    const [recommendedListings, setRecommendedListings] = React.useState([])
+    const { sendRequest } = useHttpClient();
+    const { isLoggedIn, profile } = React.useContext(UserLoginContext);
+    React.useEffect(() => {
+        const getSellerItems = async () => {
+            let items = await sendRequest(LISTING, "GET", {});
+            items = buildListings(items);
+            setListings(items);
+        };
+        
+        const getRecommendedItems = async () => {
+            if (isLoggedIn) {
+                let recommendations = await sendRequest(RECOMMENDATION,"GET", {}, `/${profile.id}`);
+                recommendations = buildListings(recommendations);
+                setRecommendedListings(recommendations)
+            }
+        };
+        getRecommendedItems();
+        getSellerItems();
+    }, [sendRequest, isLoggedIn, profile]);
+    return (
+        <React.Fragment>
+            <PageHeader headerTitle='Listing'>
+                <Accordion>
+                    <AccordionSummary
+                        expandIcon={<ExpandMoreIcon />}
+                        aria-controls="panel1a-content"
+                        id="panel1a-header"
+                    >
+                        <Typography gutterBottom variant="h3" align='center'>
+                            Recommendations
+                        </Typography>
+                    </AccordionSummary>
+                    <AccordionDetails>
+                        <RecommendedCarousel listings={recommendedListings} />
+                    </AccordionDetails>
+                </Accordion>
+                <Accordion>
+                    <AccordionSummary
+                        expandIcon={<ExpandMoreIcon />}
+                        aria-controls="panel1a-content"
+                        id="panel1a-header"
+                    >
+                        <Typography gutterBottom variant="h3" align='center'>
+                            Listings
+                        </Typography>
+                    </AccordionSummary>
+                    <AccordionDetails>
+                        <ListingGrid listings={listings} />
+                    </AccordionDetails>
```

Comment:
Can you default both of these to be open?

Reply:
fixed